test/language/expressions/arrow-function/dstr-ary-init-iter-close.js

```diff
@@ -0,0 +1,73 @@
+// This file was procedurally generated from the following sources:
+// - src/dstr-binding/ary-init-iter-close.case
+// - src/dstr-binding/default/arrow-function.template
+/*---
+description: Iterator is closed when not exhausted by pattern evaluation (arrow function expression)
+esid: sec-arrow-function-definitions-runtime-semantics-evaluation
+es6id: 14.2.16
+features: [Symbol.iterator]
+
+flags: [generated]
+info: >
+    ArrowFunction : ArrowParameters => ConciseBody
+    
+    [...]
+    4. Let closure be FunctionCreate(Arrow, parameters, ConciseBody, scope, strict).
+    [...]
+    
+    9.2.1 [[Call]] ( thisArgument, argumentsList)
+    
+    [...]
+    7. Let result be OrdinaryCallEvaluateBody(F, argumentsList).
+    [...]
+    
+    9.2.1.3 OrdinaryCallEvaluateBody ( F, argumentsList )
+    
+    1. Let status be FunctionDeclarationInstantiation(F, argumentsList).
+    [...]
+    
+    9.2.12 FunctionDeclarationInstantiation(func, argumentsList)
+    
+    [...]
+    23. Let iteratorRecord be Record {[[iterator]]:
+        CreateListIterator(argumentsList), [[done]]: false}.
+    24. If hasDuplicates is true, then
+        [...]
+    25. Else,
+        b. Let formalStatus be IteratorBindingInitialization for formals with
+           iteratorRecord and env as arguments.
+    [...]
+
+    13.3.3.5 Runtime Semantics: BindingInitialization
+    
+    BindingPattern : ArrayBindingPattern
+    
+    [...]
+    4. If iteratorRecord.[[done]] is false, return ? IteratorClose(iterator,
+       result).
+    [...]
+    
+---*/
```

Comment:
the generation tool is creating some noise: 

- trailing whitespace on empty lines (from the template and case files)
- adding an extra blank line (8 to 10 and 48 to 50)
- `info: |` converted to `info: >`